letheql/query.go

```diff
@@ -52,53 +51,57 @@ type TimeRange struct {
 	End   time.Time
 }
 
-func ProcQuery(query string, timeRange TimeRange) (QueryData, error) {
+type Query interface {
+	Exec() []string
+}
 
-	fmt.Printf("ProcQuery: query=%s, timeRange=%s\n", query, timeRange)
-	ok, filterType, err := filter.IsFilterExist(query)
-	if err != nil {
-		return QueryData{}, err
-	}
+type query struct {
+	q       string
+	filter  filter.Filter
+	keyword string
+	engine  *Engine
+}
 
-	var parsableQuery, keyword string
-	var f filter.Filter
-	if ok {
-		parts := strings.Split(query, filterType)
-		parsableQuery = strings.TrimSpace(parts[0])
-		keyword = strings.TrimSpace(parts[1])
-		filterFromQuery, err := filter.FilterFromQuery(query)
-		if err != nil {
-			return QueryData{}, err
-		}
-		f = filterFromQuery
-	} else {
-		parsableQuery = query
-		keyword = ""
-	}
+func (q *query) String() string {
+	return q.q
+}
 
-	if len(query) < 1 {
-		return QueryData{}, errors.New("empty query")
+func (q *query) Exec() []string {
+
+	// q.engine.exec()
+	// to do
+	return nil
+}
+
+func ProcQuery(queryString string, timeRange TimeRange) (QueryData, error) {
+
+	log.Printf("ProcQuery: queryString=%s, timeRange=%s\n", queryString, timeRange)
+	engine := &Engine{}
+	query, err := engine.newQuery(queryString)
+	if err != nil {
+		return QueryData{}, nil
 	}
 
-	expr, err := metricsql.Parse(parsableQuery)
+	expr, err := metricsql.Parse(query.q)
 	if err != nil {
-		log.Println("ProcQuery: Parse: err=", err)
+		log.Printf("parse query failed. err: %v\n", err)
 		return QueryData{}, err
```

Comment:
```suggestion
		return QueryData{}, fmt.Errorf("cannot parse %w",err)
```

이런 식으로 표준화하는 게 좋을 것 같습니다.
당장 중요한 건 아니니까 다음에 해도 될 것 같고요.

Reply:
네 `log`패키지를 이용한 코드는 제거하고 모두 호출부에서 error 발생시 한번에 로그를 남기는 것이 좋겠네요